core/jazz_slack-channel/index.js

```diff
@@ -0,0 +1,504 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+/**
+Nodejs Template Project
+@author:
+@version: 1.0
+ **/
+
+const errorHandlerModule = require("./components/error-handler.js"); //Import the error codes module.
+const responseObj = require("./components/response.js"); //Import the response module.
+const configObj = require("./components/config.js"); //Import the environment data.
+const logger = require("./components/logger.js")();
+const request = require("request");
+
+module.exports.handler = (event, context, cb) => {
+
+  //Initializations
+  let errorHandler = errorHandlerModule(),
+  config = configObj.getConfig(event, context);
+  logger.init(event, context);
+  let isServAccRequested = false;
+
+  genericInputValidation(event, config)
+    .then(() => createPublicSlackChannel(config, event.body))
+    .then(res => addMemberToChannel(config, res.data, isServAccRequested, event.body))
+    .then(res => {
+      logger.info("successfully created public slack channel: " + JSON.stringify(res));
+      let result = {
+        "channel_id": res.data.channelInfo.id,
+        "channel_name": res.data.channelInfo.name,
+        "channel_link": res.data.channelInfo.link,
+        "members": res.data.slackChannelMembers,
+      };
+      return cb(null, responseObj(result, event.body));
+    })
+    .catch(error => {
+      logger.error(error);
+      if (error.result === "inputError") {
+        return cb(JSON.stringify(errorHandler.throwInputValidationError(error.message)));
+      } else if (error.result === "unauthorized") {
+        return cb(JSON.stringify(errorHandler.throwUnauthorizedError(error.message)));
+      } else {
+        return cb(JSON.stringify(errorHandler.throwInternalServerError(error.message)));
+      }
+    });
+};
+
+function genericInputValidation(event, config) {
+  logger.debug("Inside genericInputValidation");
+  return new Promise((resolve, reject) => {
+    if (!event || !event.body || !event.method) {
+      reject({
+        result: "inputError",
+        message: "Service inputs not defined!"
+      });
+    }
+
+    if (!event.body.channel_name) {
+      reject({
+        result: "inputError",
+        message: "channel_name is required!."
```

Comment:
Update message to "Channel name is required."

Reply:
Updated